### PR TITLE
Enable support for STM32U585

### PR DESCRIPTION
**Pull request by anhu**
* add a client and server over UART command using DMA.
* use kyber level 1 if HAVE_PQC is defined.
* only call TLSX_UseSupportedCurve() for kyber level 1.
* make sure only one of HAVE_LIBOQS or HAVE_PQM4 is defined.

### CODE REVIEW

File: IDE/STM32Cube/wolfssl_example.c [hunk 1]
```diff
@@ -105,10 +105,9 @@
 #endif /* WOLFSSL_STATIC_MEMORY */
 
 
-/* UART definitions */
-#ifndef HAL_CONSOLE_UART
-#define HAL_CONSOLE_UART huart4
-#endif
+/* This sets which UART to use for the console.  It is something you will have
+ * to configure in STMCubeIDE and then change here. */
+#define HAL_CONSOLE_UART huart3
```
Comment (dgarske) [inline IDE/STM32Cube/wolfssl_example.c:111]: Please add `#ifndef HAL_CONSOLE_UART` around this.

**Review by dgarske — CHANGES_REQUESTED** [review]
All looks good here. I am not seeing the changes for adding `HAVE_PQM4` in this, but I know they are there. Feel free to put up a public wolfSSL PR for this stuff. Might be easier.

File: IDE/STM32Cube/wolfssl_example.c (hunk 1 above)
Comment (anhu) [reply IDE/STM32Cube/wolfssl_example.c:111]: fixed on public PR.

**Comment by anhu** [comment]
Closing this PR as it has been re-opened as public PR.

**Pull request closed without merging**